### ISSUE TITLE
with 64-bit wine, std.fs.Dir.openDir incorrectly reports error.FileNotFound when opening a sub-directory

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#3500]
There is a bug in Wine 64 bit which prevents `std.fs.Dir.openDir` from opening a sub-directory. It works fine with 32 bit wine, however.

Run this in any directory that has sub-directories:

```zig
const std = @import("std");

pub fn main() anyerror!void {
    var it = try std.fs.walkPath(std.heap.direct_allocator, ".");
    while (try it.next()) |entry| {
        std.debug.warn("{}\n", entry.path);
    }
}
```

```
$ zig build-exe test.zig -target x86_64-windows
$ wine64 test.exe
.\.gdb_history
.\.test.zig.swp
.\android_main
.\android_main.o
.\android_main.zig
.\big-source
.\big-source.o
.\big-source.zig
error: FileNotFound
/home/andy/dev/zig/lib/std/fs.zig:790:47: 0x1400238e0 in std.fs.Dir::std.fs.Dir.openDirW (test.obj)
```

[Upstream bug report](https://bugs.winehq.org/show_bug.cgi?id=47979)

**Comment by stratact** [comment]
May I take this one?

**Comment by andrewrk** [comment]
By all means, but just to make sure you understand what this issue entails:

 * Look at the upstream bug report linked above
 * Clone the wine source code and work with the wine developers to fix the issue and get the issue fixed in wine
 * Report back when it's solved

**Comment by stratact** [comment]
Gotcha. I'll find another issue to tackle that's internal to Zig.

**Comment by andrewrk** [comment]
The upstream bug has been fixed in Wine 4.19 :tada:
